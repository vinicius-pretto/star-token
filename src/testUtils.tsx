import * as React from "react";
import { render } from "@testing-library/react";
import { Provider as ReduxProvider } from "react-redux";
import configureStore, { initialState } from "./ui/redux/store";

const Providers = ({ children, state }) => {
  return (
    <ReduxProvider store={configureStore(state)}>{children}</ReduxProvider>
  );
};

const customRender = (ui, state = initialState) => {
  const options = {
    wrapper: (props) => <Providers {...props} state={state} />,
  };
  return render(ui, options);
};

export * from "@testing-library/react";
export { customRender as render };
