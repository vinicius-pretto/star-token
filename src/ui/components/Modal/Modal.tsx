import * as React from "react";

interface Props {
  title: string;
  isOpen: boolean;
  onClose: any;
  children: any;
}

const Modal = ({ title, isOpen, onClose, children }: Props) => {
  return (
    <div id="modal" className={isOpen ? "active" : ""}>
      <div className="modal-dialog">
        <div className="modal-dialog-header">
          <h2 className="section-title">{title}</h2>
          <div
            className="icon-button"
            role="button"
            aria-label="Close"
            tabIndex={0}
            onClick={onClose}
          >
            <div className="icon icon--close"></div>
          </div>
        </div>

        <div className="modal-dialog-body">{children}</div>
      </div>
    </div>
  );
};

export default Modal;
