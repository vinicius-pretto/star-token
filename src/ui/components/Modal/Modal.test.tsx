import * as React from "react";
import { render } from "../../../testUtils";
import Modal from "./Modal";

describe("Modal", () => {
  it("snapshot", () => {
    const { asFragment } = render(
      <Modal title="Colors" isOpen={true} onClose={jest.fn()}>
        <div>Content</div>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
