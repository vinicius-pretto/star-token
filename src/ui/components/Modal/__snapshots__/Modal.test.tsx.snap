// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot 1`] = `
<DocumentFragment>
  <div
    class="active"
    id="modal"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-dialog-header"
      >
        <h2
          class="section-title"
        >
          Colors
        </h2>
        <div
          aria-label="Close"
          class="icon-button"
          role="button"
          tabindex="0"
        >
          <div
            class="icon icon--close"
          />
        </div>
      </div>
      <div
        class="modal-dialog-body"
      >
        <div>
          Content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
